test/test_torch.py

```diff
@@ -3621,6 +3621,87 @@ def test_exponential(self, device, dtype):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
+    @dtypes(*torch.testing.get_all_fp_dtypes(include_bfloat16=False, include_half=False))
+    def test_cov(self, device, dtype):
+        shapes = [(1,), (10,),
```

Comment:
This is a good start to a test but I think it can do more to programmatically enumerate the inputs by enumerating over the product of these inputs.

Reply:
I am not sure which tensors did you mean when you talk about using product of these inputs. Both the input (input1, input2) tensors need to be of the same shape else it would raise an error. Did you mean something else ? @mruberry

Reply:
Sorry, by "product" I mean by using itertools.product()